### PR TITLE
Ensuring minimal documentation for classes declared in script-engine

### PR DESCRIPTION
**Pull request by odysseus654**
Proposed for a starting point of more complex scripting documentation

### CODE REVIEW

**Comment by digisomni** [comment]
This is for Doxygen.

File: libraries/script-engine/src/AbstractScriptingServicesInterface.h [hunk 1]
```diff
@@ -8,7 +8,8 @@
 //  Distributed under the Apache License, Version 2.0.
 //  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
 //
-
+/** @addtogroup ScriptEngine */
+/*@{*/
```
Comment (ctrlaltdavid) [inline libraries/script-engine/src/AbstractScriptingServicesInterface.h:12]:
```suggestion
/** @addtogroup ScriptEngine
 *  @{
 */
```
This, along with the change at the end of the file, makes the class appear in the Modules section.

File: libraries/script-engine/src/AbstractScriptingServicesInterface.h [hunk 2]
```diff
@@ -23,3 +24,4 @@ class AbstractScriptingServicesInterface {
 
 
 #endif // hifi_AbstractScriptingServicesInterface_h
+/*@}*/
```
Comment (ctrlaltdavid) [inline libraries/script-engine/src/AbstractScriptingServicesInterface.h:27]:
```suggestion
/** @}*/
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by odysseus654** [comment]
Thank you for that.

Also, the reason I asked for the CR as this is in many ways changing our template .h file and may end up changing every .h file we have in our project at some point, so some review as to whether there are objections to this being "the way things are".  I've also heard objections that there should be linebreaks at the end of the file and will be adding those in as well.

**Comment by ctrlaltdavid** [comment]
It seems like it will be a very useful change - certainly worthwhile trying out. I'll try it out in the networking library, too, while I'm in there.

**Comment by odysseus654** [comment]
Also means that we need to ensure all classes "of note" are documented with at least one line, especially if they are subclassed by others

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-Interface-PR1223-1264887-_56379fe0.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-1264887-.tar_56379fe0.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-1264887-.tar_56379fe0.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-1264887-_56379fe0.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-1264887-_56379fe0.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-1264887-_56379fe0.exe

File: libraries/script-engine/src/DataViewPrototype.h [hunk 3]
```diff
@@ -65,4 +68,5 @@ public slots:
     bool realOffset(qint32& offset, size_t size) const;
 };
 
-#endif // hifi_DataViewPrototype_h
\ No newline at end of file
+#endif // hifi_DataViewPrototype_h
```
Comment (ctrlaltdavid) [inline libraries/script-engine/src/DataViewPrototype.h:71]: A couple of weird characters at the end of this line that would be good to get rid of.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]
I build the Doxygen and it seems to work well.

File: libraries/script-engine/src/AssetScriptingInterface.h [hunk 4]
```diff
@@ -24,6 +26,7 @@
 #include <BaseScriptEngine.h>
 #include <QtNetwork/QNetworkDiskCache>
 
+/// Provides the <code><a href="https://apidocs.vircadia.dev/Assets.html">Assets</a></code> scripting API
```
Comment (ctrlaltdavid) [inline libraries/script-engine/src/AssetScriptingInterface.h:29]: For items that have both Doxygen and JSDoc, I think that the Doxygen comments should be after JSDoc comments: the Doxygen comments will be closer to the code whereas the JSDoc comments will be more "distant" in that they interpret the code for JavaScript API users.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: libraries/script-engine/src/DataViewPrototype.h (hunk 3 above)
Comment (odysseus654) [reply libraries/script-engine/src/DataViewPrototype.h:71]: Yah I kinda noticed that browsing as well, couldn't really find a good commit to add that prune to... but this seems like a good place

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-Interface-PR1223-e41d9d4-_259cd20b.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-e41d9d4-_259cd20b.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-e41d9d4-.tar_259cd20b.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-e41d9d4-.tar_259cd20b.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-e41d9d4-_259cd20b.dmg

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-e41d9d4_259cd20b.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-Quest-Alpha-PR1223-e41d9d4_259cd20b.apk

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1223/Vircadia-PR1223-e41d9d4-_259cd20b.exe

**Review by ctrlaltdavid — APPROVED** [review]
Note: Can also use `///` form of Doxygen comments for `addtogroup` ...
```
/// @addtogroup ScriptEngine
/// @{
...
/// @}
```
This could be preferable.

**Comment by odysseus654** [comment]
I can make these changes if appropriate.  We are after all kinda creating a new coding standard here.

**Comment by ctrlaltdavid** [comment]
That would be good, I think. I'm using the `///` form in PR https://github.com/vircadia/vircadia/pull/1238 and it seems to make it more consistent. Would be especially relevant in files containing both JSDoc and Doxygen comments.

**Comment by odysseus654** [comment]
The "do not merge" label was requested by me as a means to counter the "changes approved" tag (which should really be "changes requested")

**Comment by digisomni** [comment]
The suggested fix-ups can be applied by @ctrlaltdavid in one of his docs PRs, so no worries. :)

**Pull request merged**